core/src/test/java/com/opensymphony/xwork2/interceptor/CSRFPreventionInterceptorTest.java

```diff
@@ -0,0 +1,152 @@
+
+        request.addHeader("sec-fetch-mode", "navigate");
+        request.addHeader("sec-fetch-dest", "script");
+        request.setMethod("GET");
+
+        ServletActionContext.setRequest(request);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        try{
+            interceptor.intercept(mai);
+            assert(true);
+        } catch (RuntimeException e){
+            assert(false);
+        }
+    }
+
+    public void testInValidTopLevelNavigation() throws Exception {
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+
+        for (String header : Arrays.asList("object", "embed")) {
+            request.addHeader("sec-fetch-site", "foo");
+            request.addHeader("sec-fetch-mode", "navigate");
+            request.addHeader("sec-fetch-dest", header);
+            request.setMethod("GET");
+
+            ServletActionContext.setRequest(request);
+            ActionContext context = ServletActionContext.getActionContext();
+            mai.setInvocationContext(context);
+
+            try{
+                interceptor.intercept(mai);
+                assert(false);
+            } catch (RuntimeException e){
+                assert(true);
+            }
+            request.removeHeader("sec-fetch-dest");
+        }
+    }
+
+    public void testPathInExemptedPaths() throws Exception {
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+
+        request.addHeader("sec-fetch-site", "foo");
+        request.setContextPath("/foobar");
+        interceptor.setExemptedPaths(Arrays.asList("/foobar", "/test"));
+
+        ServletActionContext.setRequest(request);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        try{
+            interceptor.intercept(mai);
+            assert(true);
+        } catch (RuntimeException e){
+            assert(false);
+        }
+    }
+
+    public void testPathNotInExemptedPaths() throws Exception {
+        MockActionInvocation mai = new MockActionInvocation();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+
+        request.addHeader("sec-fetch-site", "foo");
+        request.setContextPath("/foobar");
+        interceptor.setExemptedPaths(Arrays.asList("/test"));
+
+        ServletActionContext.setRequest(request);
+        ActionContext context = ServletActionContext.getActionContext();
+        mai.setInvocationContext(context);
+
+        try{
+            interceptor.intercept(mai);
+            assert(false);
+        } catch (RuntimeException e){
+            assert(true);
+        }
+    }
+
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        interceptor = new CSRFPreventionInterceptor();
```

Comment:
let's inline this value in the field declaration and make it final, it's somewhat easier to read :)